### PR TITLE
Split grafana and prometheus into separate pods

### PR DESCRIPTION
**Pull request by darahayes**
The result of this PR is that grafana and prometheus are split into their own pods.

In addition, a new service called `prometheus-internal` was added which directly exposes the prometheus container. (no proxy). This additional service is needed so grafana can access prometheus as a datasource. This approach was outlined in the [metrics proposal](https://github.com/aerogear/proposals/pull/5/files#diff-051c99b3216ebd416c62117d4f732680R132). I'm looking for suggestions on the name though.

The datasource config file seems to be broken at the moment, so you will not see the prometheus data source show up automatically when the apb is run. You can manually add the prometheus service afterwards using the hostname `http://prometheus-internal`

I'm aware that there is still plenty of room for improvement in this APB which I think should be tackled in a separate PR.

### CODE REVIEW

**Comment by darahayes** [comment]
cc @wtrocki

**Comment by wtrocki** [comment]
@darahayes Verified locally. It works like dream for general dashboard.

File: roles/provision-prometheus-apb/tasks/main.yml [hunk 1]
```diff
@@ -101,6 +101,31 @@
...
+    volumes:
+      - name: '{{ prometheus_configmap_volume_name }}'
+        config_map:
+          defaultMode: 420
+          name: '{{ prometheus_configmap_name }}'
+      - name: prometheus-volume-1
+        persistent_volume_claim:
+          claim_name: prometheus-claim
+
+- name: create grafana deployment config
```
Comment (paolobueno) [inline roles/provision-prometheus-apb/tasks/main.yml]: Would it be interesting to split these into a separate `provision-grafana-apb` role?

Comment (darahayes) [reply roles/provision-prometheus-apb/tasks/main.yml]: @paolobueno yes this is a great suggestion!

**Comment by darahayes** [comment]
@paolobueno suggested splitting the grafana and prometheus provisioning into separate roles. While I initially agreed with this approach, I did some investigation and I've concluded there's no benefit in doing this for the time being.

The [ansibleplaybookorg](https://github.com/ansibleplaybookbundle) has a number of example APBs that provision multiple services e.g. (Application + Database) and they seem to keep them in the same roles.

- [HA Wordpress](https://github.com/ansibleplaybookbundle/wordpress-ha-apb)
- [Rocket Chat](https://github.com/ansibleplaybookbundle/rocketchat-apb/tree/master/rhcc/roles)
- [hello world persistent db example](https://github.com/ansibleplaybookbundle/hello-world-db-apb)

@paolobueno @wtrocki what do you think?

**Comment by wtrocki** [comment]
@darahayes Do you need me to verify that again?

**Comment by darahayes** [comment]
Yes please @wtrocki . Now that the name has been changed I think it would be wise to reverify.

**Comment by wtrocki** [comment]
@paolobueno Can you verify that as well? I'm happy to merge (already approved this change)

**Comment by paolobueno** [comment]
@wtrocki tried verfiying but blocked on my local origin cluster: https://gist.github.com/paolobueno/c3430125b2ab43f22884597f6314b29d

Feel free to go ahead and merge!

**Comment by wtrocki** [comment]
@paolobueno Looks like missing selinux options for openshift containers.

File: roles/provision-metrics-apb/templates/grafana-data-sources.yml.j2 [hunk 2]
```diff
@@ -5,7 +5,7 @@ datasources:
   # <string, required> access mode. direct or proxy. Required
   access: proxy
   # <string> url
-  url: http://localhost:9090
+  url: http://prometheus-internal
```
Comment (pb82) [inline roles/provision-metrics-apb/templates/grafana-data-sources.yml.j2]: is it this URL because of the service name?

Comment (wtrocki) [reply roles/provision-metrics-apb/templates/grafana-data-sources.yml.j2]: Yes. We needed to use service name. 
This service will be always in the same namespace so we can use shorter form.

**Comment by wtrocki** [comment]
@pb82 thanks for help with verification. Let's merge this.

**Pull request merged**